plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -918,6 +934,28 @@ public static Map<String, String> createTableProperties(ConnectorTableMetadata t
                 .buildOrThrow();
     }
 
+    private static Optional<String> toAvroCompressionCodecTableProperty(HiveCompressionCodec compressionCodec)
+    {
+        return switch (compressionCodec) {
+            case NONE -> Optional.of("uncompressed");
+            case SNAPPY -> Optional.of("snappy");
+            case LZ4 -> Optional.empty(); // writing to such table is unsupported by Trino
```

Comment:
nit: you may as well throw here 
`new TrinoException(NOT_SUPPORTED, "Compression codec LZ4 not supported for AVRO")`

and return `String` instead of `Optional<String>`